llvm/utils/TableGen/Common/CodeGenRegisters.h

```diff
@@ -185,7 +185,7 @@ class CodeGenRegister {
 
   CodeGenRegister(const Record *R, unsigned Enum);
 
-  StringRef getName() const;
+  StringRef getName() const { return TheDef->getName(); }
```

Comment:
The original asserted/validated `TheDef`, should it be here too?

Reply:
Reinstated. I'm not sure if such asserts are really useful, but that's for another day to debate.